main.rb

```diff
@@ -0,0 +1,139 @@
+module Enumerable
+  # 1.my_each
+  def my_each
+    for item in self
+      yield(item)
+    end
+  end
+
+  # 2.my_each_with_index
+  def my_each_with_index
+    index = 0
+    my_each do |item|
+      yield(item, index)
+      index += 1
+    end
+  end
+
+  # 3.my_select
+  def my_select
+    select_arr_result = []
+    each do |item|
+      select_arr_result << item if yield(item)
+    end
+    select_arr_result
+  end
```

Comment:
- [ ] These three methods all should return an `Enumerator` when no block is given